app/pages/[teamSlug]/team/[userId].page.tsx

```diff
@@ -48,60 +47,73 @@ const UserDetailsPage = (): JSX.Element => {
-            {data?.user.projects.map((project) => (
-              <li key={project.id}> {project.title}</li>
-            ))}
-          </ul>
-          <h1 className="text-xl font-semibold text-gray-400"> All Projects:</h1>
+            <div className="w-full">
+              <h1 className="text-4xl dark:text-blue-400">{data?.user.name}</h1>
 
-          <ul>
-            {allProjects?.projects.map((project) => (
-              <li key={project.id} className="p-3">
-                <span className=" inline-block w-32"> {project.title} </span>
-                <Toggle
-                  disabled={!isAdmin}
-                  checked={data?.user.projects.some((userProject) => userProject.id === project.id) ?? false}
-                  onChange={(newValue) => {
-                    if (!data?.user) {
-                      return
-                    }
-                    if (newValue === false) {
-                      deleteProjectMembership({ projectID: project.id, userID: data?.user.id })
-                    } else {
-                      createProjectMembership({ projectID: project.id, userID: data?.user.id })
-                    }
-                  }}
-                />
-              </li>
-            ))}
-          </ul>
-        </article>
-        {userData?.user.role === 'ADMIN' && (
-          <div className="flex justify-between pt-20">
-            <div className="flex justify-center">
-              {data?.user.role === 'MEMBER' && (
-                <Button className="mx-4" variant="primary" onClick={handleUpgradeClick}>
-                  Upgrade
-                </Button>
-              )}
-              {data?.user.role === 'ADMIN' && (
-                <Button variant="secondary" onClick={handleDowngradeClick}>
-                  Downgrade
-                </Button>
-              )}
-            </div>
-            <div>
-              <Button variant="danger">Kick from Team</Button>
+              <div className="my-1  border border-gray-400 dark:border-blue-400 " />
+
+              <div className="flex justify-between">
+                <h1 className="text-xl font-semibold text-gray-400">Team member</h1>
+
+                <div className="flex">
+                  {isAdmin && (
+                    <>
+                      {data?.user.role === Role.Member ? (
+                        <Button className="mr-4" variant="secondary" onClick={handleUpgradeClick}>
+                          Promote to admin
+                        </Button>
+                      ) : (
+                        <Button className="mr-4" variant="secondary" onClick={handleDowngradeClick}>
+                          Demote to member
+                        </Button>
+                      )}
+                    </>
+                  )}
+
+                  <Button variant="danger">Remove from team</Button>
+                </div>
+              </div>
             </div>
           </div>
-        )}
+          {isAdmin ? (
```

Comment:
How about moving the `isAdmin` check inside the `.map()` method?
This means we only have to show/hide the toggle and not the whole list.

@WINt1703, what do you think?

Reply:
Never mind, I just now understood that these are two completely different lists. 😄